algorithms/kernel/elastic_net/elastic_net_train_dense_default_impl.i

```diff
@@ -0,0 +1,288 @@
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+/*
+//++
+//  Implementation of auxiliary functions for elastic net classification
+//  (defaultDense) method.
+//--
+*/
+
+#ifndef __ELASTIC_NET_TRAIN_DENSE_DEFAULT_IMPL_I__
+#define __ELASTIC_NET_TRAIN_DENSE_DEFAULT_IMPL_I__
+
+#include "elastic_net_train_kernel.h"
+#include "elastic_net_model_impl.h"
+#include "service_error_handling.h"
+#include "service_algo_utils.h"
+#include "algorithms/optimization_solver/objective_function/mse_batch.h"
+#include "algorithms/optimization_solver/coordinate_descent/coordinate_descent_batch.h"
+
+#include "service_numeric_table.h"
+#include "service_math.h"
+
+using namespace daal::algorithms::elastic_net::training::internal;
+using namespace daal::algorithms::optimization_solver;
+using namespace daal;
+
+namespace daal
+{
+namespace algorithms
+{
+namespace elastic_net
+{
+namespace training
+{
+namespace internal
+{
+//////////////////////////////////////////////////////////////////////////////////////////
+// TrainBatchKernel
+//////////////////////////////////////////////////////////////////////////////////////////
+template <typename algorithmFPType, elastic_net::training::Method method, CpuType cpu>
+services::Status TrainBatchKernel<algorithmFPType, method, cpu>::compute(
+    const HostAppIfacePtr & pHost, const NumericTablePtr & x, const NumericTablePtr & y, elastic_net::Model & m, Result & res,
+    const Parameter & par, services::SharedPtr<daal::algorithms::optimization_solver::mse::Batch<algorithmFPType> > & objFunc)
+{
+    services::Status s;
+    const size_t nFeatures           = x->getNumberOfColumns();
+    const size_t nRows               = x->getNumberOfRows();
+    const size_t p                   = nFeatures + 1;
+    const size_t nDependentVariables = m.getBeta()->getNumberOfRows();
+    DAAL_ASSERT(p == m.getNumberOfBetas());
+
+    algorithmFPType * xMeansPtr;
+    daal::internal::TArray<algorithmFPType, cpu> xMeans;
+
+    algorithmFPType * yMeansPtr;
+    daal::internal::TArray<algorithmFPType, cpu> yMeans;
+
+    NumericTablePtr xTrain = x;
+    NumericTablePtr yTrain = y;
+    if (par.interceptFlag == true)
+    {
+        if (par.dataUseInComputation == doNotUse)
+        {
+            int result = 0;
+            xTrain     = daal::internal::HomogenNumericTableCPU<algorithmFPType, cpu>::create(nFeatures, nRows, &s);
```

Comment:
We could try to re-use the date and return it after computation.

Reply:
it'll be done on optimization stage in future